src/jit/hwintrinsicxarch.cpp

```diff
@@ -781,9 +781,59 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
     GenTree*  retNode  = nullptr;
     GenTree*  op1      = nullptr;
     GenTree*  op2      = nullptr;
+    int       ival     = -1;
+    int       simdSize = simdSizeOfHWIntrinsic(intrinsic, sig);
     var_types baseType = TYP_UNKNOWN;
+    var_types retType  = TYP_UNKNOWN;
+
     switch (intrinsic)
     {
+        case NI_SSE2_CompareLessThan:
+            assert(sig->numArgs == 2);
+            op2      = impSIMDPopStack(TYP_SIMD16);
+            op1      = impSIMDPopStack(TYP_SIMD16);
+            baseType = getBaseTypeOfSIMDType(sig->retTypeSigClass);
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, intrinsic, baseType, simdSize);
+                    break;
+                case TYP_BYTE:
+                case TYP_SHORT:
+                case TYP_INT:
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op2, op1, intrinsic, baseType, simdSize);
+                    break;
+                default:
+                    retNode = gtNewMustThrowException(CORINFO_HELP_THROW_ARGUMENTEXCEPTION, JITtype2varType(sig->retType),
```

Comment:
You do not need the throwing.

Reply:
Can you elaborate?

Reply:
This is a non-generic intrinsic, so the type-argunment is already checked by the type system. In other words, if you call `CompareLessThan` on `Vector128<long>`, .NET Core would reject it and say "no method found".

Reply:
> is already checked by the type system

You mean C# compiler or CLR runtime?

Reply:
Both.